src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1035,6 +1035,15 @@
     ["SELECT timestamp FROM factsets WHERE certname=? ORDER BY timestamp DESC" certname]
     (:timestamp (first result-set))))
 
+
+(pls/defn-validated factset-producer-timestamp :- (s/maybe pls/Timestamp)
+  "Return the factset producer-timestamp for the given certname, nil if not found"
+  [certname :- String]
+  (sql/with-query-results result-set
+    ["SELECT producer_timestamp FROM factsets WHERE certname=? ORDER BY producer_timestamp DESC" certname]
```

Comment:
This is a bald imitation of factset-timestamp above. Did some spelunking to see if I could figure out why these queries were ever sorted... they came about in 71c1af56247337e16d680ef357360651af57fd16 in more or less this form. Not sure what that tells us...